### PR TITLE
Add --no-suffix, --minitest options and FormObject Pattern

### PR DESCRIPTION
**Pull request by caioeps**
### Added
- `--minitest` option to specify you're using minitest.
- `--rspec` option to specify you're using minitest. It is the default behaviour.
- `--no-suffix` option to specify you don't want any suffixes.
### Changed
- Get back to using suffixes by default.
### Removed
- `--test-suite` option to specify which test suite you're using.

### Added
- Form pattern generator

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+## [0.2.0] - 2017-08-07
+### Added
+- `--minitest` option to specify you're using minitest.
+- `--rspec` option to specify you're using minitest. It is the default behaviour.
```
Comment (sungwoncho) [inline CHANGELOG.md]: Did you mean rspec?

File: lib/generators/base_generator.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+# Base class for all generators to come. If you want to override some behaviour, all you
+# got to do is override a method in the child class.
+class BaseGenerator < Rails::Generators::NamedBase
+  class_option :rspec, type: :boolean, desc: 'DEFAULT: Define rspec as your test framework.'
+  class_option :minitest, type: :boolean, desc: 'Test framework to generate test. (rspec or minitest)'
+  class_option :no_suffix, type: :boolean, desc: "If you don't want any pattern suffix. Ex: AuthenticationService. Service is the suffix."
+
+  # Creates the file for the requested pattern based on a template file.
+  def copy_service_file
```
Comment (sungwoncho) [inline lib/generators/base_generator.rb]: Is this used anywhere? Feels a bit odd that a method for a 'service' is implemented in a generic `BaseGenerator`.

File: lib/generators/base_generator.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    template "#{pattern_name}.rb", generated_file_path
+  end
+
+  # Creates the test file for the request pattern based on a template file.
+  def copy_forms_test_file
```
Comment (sungwoncho) [inline lib/generators/base_generator.rb]: Similar concern as above.

**Review by sungwoncho — COMMENTED** [review]
Looking great so far. Left some comments.

**Comment by sungwoncho** [comment]
Also test seems to be not passing.

**Comment by caioeps** [comment]
Tests were passing. Apparently there was some problem with the CI.
I corrected the problems you pointed out.

**Comment by sungwoncho** [comment]
Released in [v0.1.0](https://github.com/sungwoncho/pattern_generator/blob/master/CHANGELOG.md#010---2017-08-07). Thanks for your contribution.

You're now a collaborator on GitHub and rubygems.org.

**Pull request merged**